content/ui-patterns/saving.mdx

```diff
@@ -0,0 +1,266 @@
+</Box>
+
+#### Bottom-right
+
+If there is a button to submit and a button to cancel, the button to cancel should go to the **left** of the submit button.
+
+<Box mb={3} display="flex" alignItems="flex-start" flexDirection={["column", "column", "column", "column", "row"]} sx={{gap: 3}}>
+    <img
+        width="456"
+        alt="A small form in a dialog that has 'Cancel' and 'Save' buttons at the bottom right"
+        src="https://user-images.githubusercontent.com/2313998/151623724-d59d5967-5484-4e81-a4c2-1109409956ee.png"
+    />
+    <Box as="p" mt="0">If the save button is used in a dialog, place the button at the bottom right of the dialog. Right-aligned dialog buttons feel familiar because it's a common pattern across various platforms such as Windows, macOS, and Android.</Box>
+</Box>
+
+<Box mb={3} display="flex" alignItems="flex-start" flexDirection={["column", "column", "column", "column", "row"]} sx={{gap: 3}}>
+    <img
+        width="456"
+        alt="A multi-line text input with 'Close with Comment' and 'Comment' buttons right-aligned below the input"
+        src="https://user-images.githubusercontent.com/2313998/151623722-96b6f25d-d8b7-4de7-affe-cd202d6282f1.png"
+    />
+    <Box as="p" mt="0">
+        If the button is used to send a message such as a comment on an issue, right-align the button below the text area. This is where GitHub has traditionally placed the <strong>Submit</strong> button for issues and pull requests, and it's common practice in other apps to place a <strong>Send</strong> button to the right.
+    </Box>
+</Box>
+
+
+<!--
+TODO:
+1. Be more specific about when a button should be placed somewhere besides the bottom-right. Define alternative patterns and when to use them.
+
+2. Consider how we might define more patterns to keep the "Save" button easily accessible on long forms:
+- add save and cancel buttons on top AND bottom?
+- use a sticky header with save and cancel buttons?
+-->
+
+#### Adjacent to inline editing input
+
+To simplify the interface, some content may be edited inline without taking the user to a separate flow.
+
+An input used for inline editing should have a save button placed very close to visually connect the input with the button.
+
+<img
+    width="960"
+    alt="Two images of a pull request header. The first image shows the title as read-only text. The second image shows a text input to edit the pull request title with 'Cancel' and 'Save' buttons below the input."
+    src="https://user-images.githubusercontent.com/2313998/151623718-9f3f7fe5-ad35-4e4a-b236-da594388203b.png"
+/>
+
+#### Other
+
+If you believe your submit button will not be intuitive or easily accessible using any of the placement options listed here, you may consider alternatives and optionally propose a new Primer pattern.
+
+## Explicit saving
```

Comment:
Just checking — are `calls to action` a form of `explicit saving`? Based on what I'm reading, I'd assume so, but they have the same hierarchy 🤔. Calls to action are also higher up in the page, which suggests they are unrelated concepts.

Reply:
Great point! I'll bring "Calls to action" into the "Explicit saving" section.